src/primitives/relocatable.ts

```diff
@@ -1,147 +1,77 @@
 import { Felt } from './felt';
-import { Uint32, UnsignedInteger } from './uint';
+import { UnsignedInteger } from './uint';
 import {
   ForbiddenOperation,
-  OffsetOverflow,
   OffsetUnderflow,
   PrimitiveError,
   SegmentError,
-} from 'result/primitives';
-import { Err, Result, Ok } from 'result/result';
+} from 'errors/primitives';
 
 export type MaybeRelocatable = Relocatable | Felt;
 
 export class Relocatable {
-  private segmentIndex: Uint32;
-  private offset: Uint32;
+  private segmentIndex: number;
+  private offset: number;
 
   constructor(segmentIndex: number, offset: number) {
-    const { value: segmentUint, error: indexErr } =
-      UnsignedInteger.toUint32(segmentIndex);
-    const { value: offsetUint, error: offsetErr } =
-      UnsignedInteger.toUint32(offset);
-    if (indexErr !== undefined) {
-      throw indexErr;
-    }
-    if (offsetErr !== undefined) {
-      throw offsetErr;
-    }
-    this.segmentIndex = segmentUint;
-    this.offset = offsetUint;
+    UnsignedInteger.ensureUint32(segmentIndex);
+    UnsignedInteger.ensureUint32(offset);
+    this.segmentIndex = segmentIndex;
+    this.offset = offset;
   }
 
-  add(other: Relocatable): Err;
-  add(other: Felt): Result<Relocatable>;
-  add(other: Uint32): Ok<Relocatable>;
-  add(other: MaybeRelocatable): Result<MaybeRelocatable>;
-
-  add(other: MaybeRelocatable | Uint32): Result<MaybeRelocatable> {
+  add(other: Felt): Relocatable;
+  add(other: number): Relocatable;
+  add(other: Relocatable): never;
+  add(other: MaybeRelocatable): MaybeRelocatable;
+  add(other: MaybeRelocatable | number): MaybeRelocatable {
```

Comment:
maybe a ts question, but it seeems like the last signature is redundant?

Reply:
> maybe a ts question, but it seeems like the last signature is redundant?

the last signature is the implementation, it  must be compatible with all of the above, so its a big union